erts/emulator/nifs/common/prim_file_nif_dyncall.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * %CopyrightBegin%
+ *
+ * Copyright Ericsson 2021. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * %CopyrightEnd%
+ */
+
+enum prim_file_nif_dyncall_op {
+    prim_file_nif_dyncall_dup,
+};
+
+struct prim_file_nif_dyncall {
+    size_t size;
+    enum prim_file_nif_dyncall_op op;
+    int completed;
+};
+
+struct prim_file_nif_dyncall_dup {
+    struct prim_file_nif_dyncall common;
+
+    int error;
+    ERL_NIF_TERM handle;
```

Comment:
Why is the file descriptor not returned as-is. Why the binary term packaging?

Reply:
I tried to find a way to declare a type for it that would work both on all Unixes and Windows, but could not figure out how to make prim_socket_nif.c, prim_file_nif_dyncall.h, unix_prim_file.c and win_prim_file.c agree on it.  If that would be feasible I would prefer such a solution.

Now it is only prim_socket_nif.c and unix_prim_file.c that needs to agree, and in both those files the type of a file descriptor is obvious.

Reply:
The binary packaging is just a way to handle the unknown size of the file descriptor, since it solves memory allocation and de-allocation